src/main/java/dev/seniorjava/githubstatisticsbot/factories/CommitLinkProvidedEventFactory.java

```diff
@@ -0,0 +1,30 @@
+package dev.seniorjava.githubstatisticsbot.factories;
+
+import com.pengrad.telegrambot.model.Message;
+import com.pengrad.telegrambot.model.Update;
+import dev.seniorjava.githubstatisticsbot.events.CommitLinkProvidedEvent;
+import dev.seniorjava.githubstatisticsbot.events.Event;
+import dev.seniorjava.githubstatisticsbot.events.PullRequestLinkProvidedEvent;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+public class CommitLinkProvidedEventFactory implements EventFactory {
+
+  @Override
+  public Event create(final Update update) {
+    final Message message = update.message();
+    return new CommitLinkProvidedEvent(message.chat().id(), message.text());
+  }
+
+  @Override
+  public boolean canCreateEvent(final Update update) {
+    if (update == null || update.message() == null || update.message().replyToMessage() == null) {
+      return false;
+    }
+    final String expectedMessageStart = "Please, provide a link to a commit strictly in the following format";
```

Comment:
Вы используете эту строку в другом классе и более того проводите сравнение на строгое соответствие строк, поэтому настоятельно рекомендую вынести ее в общедоступную константу.
Так как прямой отсылки к классу `CommitChosenEventListener` здесь нет очень легко совершить ошибку поменяв значение в одном месте и упустив из виду другие места

Reply:
Let me answer you in English, please.

First of all, thank you for your time and interesting questions.
Now I'll explain the chosen solution. You're absolutely right, that this is error-prone. But I did this on purpose. As you may know, this task was done in a few hours divided by two evenings. It's quite hard to match all best practices even if so a small project. But enough excuses.
Having the current solution I was thinking to create a public constant in the [CommitChosenEventListener](http://github.com/driversti/GithubStatisticsBot/blob/c94f68e356bd880a49c3e43319b01d785711067c/src/main/java/dev/seniorjava/githubstatisticsbot/listeners/CommitChosenEventListener.java) class because this is the owner of the string. But I simply didn't find time for this. Apart from that, as you may notice, both classes lay in different packages and I'd like to avoid making public classes, methods, variables without a need. Instead of this, I prefer to narrow their visibility. One way is to group classes by functionality and make needed parts of the code `package-private`. As for now, I didn't make a decision about what way I want to choose as a final solution. Nevertheless, I will move this string to a constant to minimize possible mistakes. Thank you 🙂